server/hotrod/src/main/java/org/infinispan/server/hotrod/HotRodServer.java

```diff
@@ -166,6 +166,8 @@ public ClientListenerRegistry getClientListenerRegistry() {
    }
 
    public ClientCounterManagerNotificationManager getClientCounterNotificationManager() {
+      if (clientCounterNotificationManager == null)
+         throw new IllegalStateException("HotRod server is not fully initialized");
```

Comment:
We should use `IllegalLifecycleStateException` for all of these types of startup exceptions. The client and core have special handling for them to ignore as needed.